core/cas-server-core-services-registry/src/main/java/org/apereo/cas/services/resource/AbstractResourceBasedServiceRegistry.java

```diff
@@ -257,6 +258,11 @@ protected void removeRegisteredService(final RegisteredService service) {
             LOGGER.debug("[{}] starts with ., ignoring", fileName);
             return new ArrayList<>(0);
         }
+        if (!Arrays.asList(getExtensions()).stream().anyMatch(ext -> fileName.endsWith(ext))) {
```

Comment:
Simplified to `if (Arrays.stream(getExtensions()).noneMatch(fileName::endsWith)) {`